src/components/spreadsheet/utils/equipment-table-utils.ts

```diff
@@ -80,6 +81,93 @@ export const updateGeneratorCells = (params: CellEditingStoppedEvent) => {
     }
 };
 
+export const updateTwtCells = (params: CellEditingStoppedEvent) => {
+    const rowNode = params.node;
+    const colId = params.column.getColId();
+    const ratioRegulationTypeText =
+        params.data?.ratioTapChanger?.regulationType;
+    const phaseRegulationTypeText =
+        params.data?.phaseTapChanger?.regulationType;
+    const previousData = params.context.dataToModify;
+    switch (colId) {
+        case 'ratioTapChanger.regulationType':
+            if (ratioRegulationTypeText === REGULATION_TYPES.DISTANT.id) {
+                // This solve the problem of required on RatioRegulatingTerminal when we set regulationType to distant
+                const regulatingTerminalGenerator =
+                    getTapChangerRegulationTerminalValue(
+                        previousData?.ratioTapChanger
+                    );
+                rowNode.setDataValue(
+                    'RatioRegulatingTerminal',
+                    regulatingTerminalGenerator
+                );
+            }
+            params.api.flashCells({
+                rowNodes: [rowNode],
+                columns: [
+                    'ratioTapChanger.regulationType',
+                    'RatioRegulatingTerminal',
+                ],
+            });
+            break;
+
+        case 'phaseTapChanger.regulationType':
+            if (phaseRegulationTypeText === REGULATION_TYPES.DISTANT.id) {
+                // This solve the problem of required on PhaseRegulatingTerminal when we set regulationType to distant
+                const regulatingTerminalGenerator =
+                    getTapChangerRegulationTerminalValue(
+                        previousData?.phaseTapChanger
+                    );
+                rowNode.setDataValue(
```

Comment:
`setDataValue` doesn't work if the modified column is hidden, you can reuse the method used for shunt compensators https://github.com/gridsuite/gridstudy-app/pull/1858/files#diff-5b30d591786ed11432767e66f7a56cfb803e5a6f88ba068b0229538b5e1f07edR188
to solve this problem